### PR TITLE
Update Stepper 

### PR DESCRIPTION
**Pull request by erenjaegar77**
Move  stepper into a new reducer to see tx status across app

- [x] New Reducer/ Actions/ Selectors

- [x] Update components 

- [x] Update how manage progress bar styles

### CODE REVIEW

File: src/components/Stepper/components/Content/Content.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  const currentStepData = useAppSelector(selectStepperCurrentStepData);
+  const bridgeStatus = useAppSelector(selectBridgeStatus);
+
+  const handleClose = React.useCallback(() => {
+    dispatch(stepperActions.reset());
+    dispatch(walletActions.resetWallet());
+  }, [dispatch]);
+
+  return (
+    <>
```
Comment (seguido) [inline src/components/Stepper/components/Content/Content.tsx:107]: This logic appears weird if currentData.step is false we show the button, if it is true we also show it but only if bridgeStatus is true as well. 
`currentStepData.step !== 'bridge' || bridgeStatus === 'success' ? <button> : null`

File: src/components/Stepper/components/Title/Title.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { styles } from './styles';
+
+const useStyles = makeStyles(styles);
+
+const _Title = () => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+
+  const walletActionsStateResult = useAppSelector(state => state.user.walletActions.result);
+
```
Comment (seguido) [inline src/components/Stepper/components/Title/Title.tsx:21]: what's with all the spacing? 👀

File: src/features/data/actions/wallet-actions.tsx [hunk 3]
```diff
@@ -536,7 +533,7 @@ const exitGovVault = (vault: VaultGov) => {
     bindTransactionEvents(
       dispatch,
       transaction,
-      { spender: contractAddr, amount: rewardAmount, token },
+      { spender: contractAddr, amount: balanceAmount, token },
```
Comment (seguido) [inline src/features/data/actions/wallet-actions.tsx:536]: this no longer passes in the same amount, are you sure of what you are passing in as balance?

File: src/features/data/actions/wallet-actions.tsx [hunk 4]
```diff
@@ -613,7 +609,7 @@ const exitBoost = (boost: BoostEntity) => {
     bindTransactionEvents(
       dispatch,
       transaction,
-      { spender: contractAddr, amount: rewardAmount, token },
+      { spender: contractAddr, amount: boostAmount, token },
```
Comment (seguido) [inline src/features/data/actions/wallet-actions.tsx:612]: Same applies here

File: src/features/vault/components/BoostWidget/Stake/Stake.tsx [hunk 5]
```diff
@@ -127,125 +127,130 @@ const StakeForm = ({
...
-            </Box>
-          </CardContent>
-        </Card>
-      </div>
-      <Stepper />
-    </>
+              )
+            ) : (
+              <Button
+                className={classes.btnSubmit}
```
Comment (seguido) [inline src/features/vault/components/BoostWidget/Stake/Stake.tsx:241]: These are still a nightmare of conditions that make it really hard to follow

File: src/features/vault/components/BoostWidget/Unstake/Unstake.tsx [hunk 6]
```diff
@@ -110,115 +111,115 @@ const UnstakeForm = ({
...
-            </Box>
-          </CardContent>
-        </Card>
-      </div>
-      <Stepper />
-    </>
+              ) : (
+                <Button
+                  onClick={handleUnstake}
+                  className={classes.btnSubmit}
```
Comment (seguido) [inline src/features/vault/components/BoostWidget/Unstake/Unstake.tsx:199]: Same, this are a pain to go through

File: src/features/vault/components/MinterCards/MintBurnCard/components/Burn.tsx [hunk 7]
```diff
@@ -148,139 +149,141 @@ export const Burn = memo(function Burn({ vaultId, minterId }: MinterCardParams)
...
+      <div className={classes.customDivider}>
+        <div className={classes.line} />
+        <img
+          alt="arrowDown"
+          src={require('../../../../../../images/icons/arrowDown.svg').default}
+        />
+        <div className={classes.line} />
+      </div>
+      <div className={classes.inputContainer}>
+        <div className={classes.balances}>
```
Comment (seguido) [inline src/features/vault/components/MinterCards/MintBurnCard/components/Burn.tsx:236]: Probably not in scope here, but since it's been reformatted I see it as well haha but this is a immense component

File: src/components/Stepper/components/Content/Content.tsx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+      : currentStepData.step === 'stake'
+      ? 'Remember-Msg-Bst'
+      : '';
+
+  const hasRewards = currentStepData.extraInfo?.rewards;
+
+  const isZapOutMessage = currentStepData.extraInfo?.zap && currentStepData.step === 'withdraw';
+
+  const successMessage = hasRewards
+    ? t(`${currentStepData.step}-Success-Content`, {
```
Comment (seguido) [inline src/components/Stepper/components/Content/Content.tsx:151]: We get many conditions being checked on several components which make it super hard to know what the outcome of a give state would be

**Review by seguido — CHANGES_REQUESTED** [review]

File: src/features/vault/components/BoostWidget/Stake/Stake.tsx (hunk 5 above)
Comment (erenjaegar77) [reply src/features/vault/components/BoostWidget/Stake/Stake.tsx:241]: This, and unstake, can be avoided, due in #1089 this components are deleted

File: src/features/data/reducers/wallet/stepper.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    | 'unstake'
+    | 'mint'
+    | 'burn'
+    | 'bridge';
+  message: string;
+  action: ThunkAction<any, any, any, any>;
+  pending: boolean;
+  extraInfo?: {
+    zap?: boolean;
+    rewards?: { token: TokenEntity; amount: BigNumber };
```
Comment (erenjaegar77) [inline src/features/data/reducers/wallet/stepper.ts:27]: @seguido about the comments on wallet-actions for changing the balance  to show the correct amount on claim-unstake from boosts and claim-withdraw from gov vaults, i added this to show the correct token and balance of the rewards 
image

**Review by seguido — APPROVED** [review]
